### PR TITLE
fix: remove start event initiator moved to or created within subprocess

### LINKED ISSUE #74: Remove camunda:initiator when a Start Event is moved to a SubProcess
__Is your feature request related to a problem? Please describe.__

capture

__Describe the solution you'd like__

When a Start Event with `camunda:initiator` property is moved or created in a SubProcess, the property should be removed.

__Describe alternatives you've considered__

-

__Additional context__

Related to https://github.com/bpmn-io/bpmn-js/issues/778

https://docs.camunda.org/manual/7.15/reference/bpmn20/custom-extensions/extension-attributes/#initiator

### PR DESCRIPTION
**Pull request by smbea**
Closes #74

### CODE REVIEW

File: lib/RemoveInitiatorBehaviour.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    if (is(shape, 'bpmn:StartEvent')) {
+
+      // if subProcess becomes the new parent
+      if (!is(context.oldParent, 'bpmn:SubProcess') && (is(context.newParent, 'bpmn:SubProcess'))) {
+
+        // remove initiator property
+        modeling.updateProperties(shape, { 'camunda:initiator': undefined });
+
+      }
+    }
```
Comment (barmac) [inline lib/RemoveInitiatorBehaviour.js:31]: Let's be explicit here. The conditions for the hook are:
* The moved/created shape is a Start Event.
* The start event has `camunda:initiator` (even if it's an empty string).
* New parent is a SubProcess.

What we don't care about is the old parent. The diagram could have been broken but behavior still needs to kick in.
Also, in the current implementation we trigger an unnecessary update if the start event does not have `camunda:initiator`.

My suggestion: check all three conditions and use `modeling.updateProperties` only if all of them are met.

**Review by barmac — CHANGES_REQUESTED** [review]
I've left a comment which needs to be addressed before we merge this PR. I like the test cases; the code is clean and well structured. Great job overall :)

File: lib/RemoveInitiatorBehaviour.js (hunk 1 above)
Comment (barmac) [reply lib/RemoveInitiatorBehaviour.js:31]: So the test case which is missing is when you move a start event which has the property and is already a child of a subprocess.

**Comment by marstamm** [comment]
I noticed that the Issue is linked in the PR description, but not in the commits. This makes it harder to find the corresponding issue once the PR is merged. Please consider adding it to one of your commits as well (cf. [Definition of Done](https://github.com/bpmn-io/internal-docs/blob/master/development/DEFINITION_OF_DONE.md))

**Comment by smbea** [comment]
Changes overview:
- Added three conditions for property removal
- Added test case for when shape with `initiator` property is moved within subprocess
- Removed some `console.log`'s I accidentally left behind

- [On commits] Started adding issue link in commits' descriptions

**Comment by barmac** [comment]
Congratulations on your first contribution 🎉

**Comment by nikku** [comment]
:tada:

**Pull request merged**